### PR TITLE
[JENKINS-39161] - Remoting documentation upgrade for the 3.0 release

### PR DESCRIPTION
**Pull request by oleg-nenashev**
Plan:

- [x] UPD: Readme
- [x] UPD: Changelog
- [x] NEW: Contributing page with release creation guides
- [x] NEW: Compatibility Notes change
- [x] NEW: Protocols description page
- POSTPONED:  -NEW: Jenkins Specifics page with configuration guidelines: configuring protocols, restricting versions, using in Docker, etc.-
- [x] UPD: System Properties page

@reviewbybees @jenkinsci/code-reviewers 

https://issues.jenkins-ci.org/browse/JENKINS-39161

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -10,14 +10,49 @@ It includes: TCP-based communication protocols, data serialization, Java classlo
...
+Currently there are two supported baselines of Remoting.
+
+#### Remoting 3
+
+Remoting 3 is a new baseline introduced in Jenkins 2.27.
+
+Major changes:
+
+* Java 7 is a new target JVM, the new remoting version is not guaranteed to work properly on Java 9 and versions below Java 7
+* <code>JNLP4-connect</code> and <code>JNLP4-plaintext</code>  protocols, which offers a high-performance communication layer
```
Comment (oleg-nenashev) [inline README.md]: typo: "protocols"

File: docs/protocols.md [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+The subsequent connection is then secured using TLS. 
+
+Protocol uses non-blocking I/O wherever possible which removes the performance bottleneck of the <code>JNLP3-connect</code> protocol.
+
+### JNLP4-plaintext
+
+* Introduced in: Remoting 3.0, [JENKINS-36871](https://issues.jenkins-ci.org/browse/JENKINS-36871)
+* Use with caution
+
+The protocol is similar to <code>JNLP4-plaintext</code>,
```
Comment (stephenc) [inline docs/protocols.md]: You mean JNLP-connect

File: docs/protocols.md [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+This protocol uses <code>SSLEngine</code> to perform a TLS upgrade of the plaintext 
+  connection before any connection secrets are exchanged. 
+The subsequent connection is then secured using TLS. 
+
+Protocol uses non-blocking I/O wherever possible which removes the performance bottleneck of the <code>JNLP3-connect</code> protocol.
+
+### JNLP4-plaintext
+
+* Introduced in: Remoting 3.0, [JENKINS-36871](https://issues.jenkins-ci.org/browse/JENKINS-36871)
+* Use with caution
```
Comment (stephenc) [inline docs/protocols.md]: This protocol was developed to allow for like for like performance comparison between the original NIO engine used by JNLP2-connect and the new NIO engine used by JNLP4. As this protocol is plaintext it is not for use outside of like for like performance testing

File: docs/protocols.md [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+
+The protocol is similar to <code>JNLP4-plaintext</code>, 
+  but it does not setup the TLS encryption between agent and master.
+In particular cases it may expose the secret data.
+
+On the other had, the protocol allows to get extra performance in the case when the remoting layer does not need its own encryption
```
Comment (stephenc) [inline docs/protocols.md]: Remove this text. The protocol is only to enable like for like performance testing and to allow for focused performance tuning of the stack outside of the performance tuning complexity of SSLEngine that the TLS layer introduces, i.e. By removing the TLS layer we can amplify the effect of IO tuning on system as we have reduced the CPU load

**Comment by oleg-nenashev** [comment]
@reviewbybees @stephenc Ready for the final review

File: docs/remoting-3-compatibility.md [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+
+Formally Remoting <code>3.x</code> is not binary compatible with Remoting <code>2.x</code>,
+  but the scope of changes is limited to a single change with a limited impact.
+
+Protocol class hierarchy change:
+
+* `JnlpProtocol` / `JnlpProtocol1` / `JnlpProtocol2` / `JnlpProtocol3` classes were removed.
+  Their functionality has been moved to the <code>JnlpProtocolHandler</code> class and its subclasses.  
+* <code>JnlpServerHandshake</code> class and all its subclasses have been removed from the library
+* Several debendent internal classes have been removed or modified
```
Comment (stephenc) [inline docs/remoting-3-compatibility.md]: Dependent is not spelled with a B

**Comment by oleg-nenashev** [comment]
@reviewbybees @jenkinsci/code-reviewers I'm using my maintainer superpower to merge this PR in order to add links to the Jenkins changelog. This PR does not impact the code && does not go into the release immediately. If you see any other issues in the text, please comment an I will follow-up in other pull requests

**Pull request merged**